stac_fastapi/pgstac/core.py

```diff
@@ -278,6 +278,9 @@ async def item_collection(
         # If collection does not exist, NotFoundError wil be raised
         await self.get_collection(collection_id, request=request)
 
+        if datetime:
+            datetime = format_datetime_range(datetime)
```

Comment:
why do we need this now?

Reply:
I see, `datetime` is automatically converted to `intervals` by https://github.com/stac-utils/stac-fastapi/blob/main/stac_fastapi/types/stac_fastapi/types/search.py#L63 or https://github.com/stac-utils/stac-fastapi/blob/main/stac_fastapi/api/stac_fastapi/api/models.py#L110

but her we want to pass a `string` to `self.post_request_model`. But `BaseSearchPostRequest` should be able to work with either with a string or tuple of datetime object ...

Reply:
I did something similar in stac-fastapi-es testing out stac-fastapi v3.0.0. I did the same thing here because test(s) were failing without it. I'm running the tests again to jog my memory.

Reply:
tests/api/test_api.py::test_item_collection_filter_datetime[api_client3] Datetime:  (datetime.datetime(2020, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), None)
FAILED

Reply:
I'm printing out Datetime at the beginning of def item_collection.

Reply:
The stac-pydantic Search model is expecting `datetime` to a string.     
```
async def item_collection(
        self,
        collection_id: str,
        request: Request,
        bbox: Optional[BBox] = None,
        datetime: Optional[DateTimeType] = None,
        limit: Optional[int] = None,
        token: Optional[str] = None,
        **kwargs,
    ) -> ItemCollection:
        """Get all items from a specific collection.
    
        Called with `GET /collections/{collection_id}/items`
    
        Args:
            collection_id: id of the collection.
            limit: number of items to return.
            token: pagination token.
    
        Returns:
            An ItemCollection.
        """
        # If collection does not exist, NotFoundError wil be raised
        await self.get_collection(collection_id, request=request)
    
        print("Datetime: ", datetime)
    
        # if datetime:
        #     datetime = format_datetime_range(datetime)
    
        base_args = {
            "collections": [collection_id],
            "bbox": bbox,
            "datetime": datetime,
            "limit": limit,
            "token": token,
        }
    
        clean = {}
        for k, v in base_args.items():
            if v is not None and v != []:
                clean[k] = v
    
>       search_request = self.post_request_model(
            **clean,
        )
E       pydantic_core._pydantic_core.ValidationError: 1 validation error for SearchPostRequest
E       datetime
E         Input should be a valid string [type=string_type, input_value=(datetime.datetime(2020, ...ime.timezone.utc), None), input_type=tuple]
E           For further information visit https://errors.pydantic.dev/2.7/v/string_type

stac_fastapi/pgstac/core.py:299: ValidationError
--------------------------------------------------------- Captured log call ----------------------------------------------------------
ERROR    stac_fastapi.api.errors:errors.py:61 1 validation error for SearchPostRequest
datetime
  Input should be a valid string [type=string_type, input_value=(datetime.datetime(2020, ...ime.timezone.utc), None), input_type=tuple]
    For further information visit https://errors.pydantic.dev/2.7/v/string_type
NoneType: None

Reply:
this is core.py:299: 
```      
search_request = self.post_request_model(
            **clean,
        )

Reply:
https://github.com/stac-utils/stac-pydantic/blob/79248a2ecac2eff8490dc7ca15131336a16f24f9/stac_pydantic/api/search.py#L32-L44

Reply:
This is for this test: FAILED tests/api/test_api.py::test_item_collection_filter_datetime